src/com/google/guiceberry/junit4/GuiceBerryRule.java

```diff
@@ -16,40 +16,54 @@
 package com.google.guiceberry.junit4;
 
 import com.google.guiceberry.DefaultEnvSelector;
-import com.google.guiceberry.GuiceBerryEnvSelector;
 import com.google.guiceberry.GuiceBerry;
 import com.google.guiceberry.GuiceBerry.GuiceBerryWrapper;
+import com.google.guiceberry.GuiceBerryEnvSelector;
 import com.google.guiceberry.TestDescription;
 import com.google.inject.Module;
-
 import org.junit.rules.MethodRule;
+import org.junit.rules.TestRule;
 import org.junit.runners.model.FrameworkMethod;
 import org.junit.runners.model.Statement;
 
 /**
  * {@link GuiceBerry} adapter for JUnit4 tests.
- * 
+ *
+ * <p>This is the preferred implementation to use, as it does not depend on the client code
+ * supplying the right object for {@code Object target}.
+ *
+ * <p>@see If you are looking for a {@link TestRule}, see {@link GuiceBerryTestRule}.
+ *
  * @author Luiz-Otavio "Z" Zorzella
  */
 public class GuiceBerryRule implements MethodRule {
 
-  private final GuiceBerryEnvSelector guiceBerryEnvSelector;
+  private GuiceBerryEnvSelector guiceBerryEnvSelector;
 
   public GuiceBerryRule(Class<? extends Module> envClass) {
-    this.guiceBerryEnvSelector = DefaultEnvSelector.of(envClass);
+    this(DefaultEnvSelector.of(envClass));
   }
 
   public GuiceBerryRule(GuiceBerryEnvSelector guiceBerryEnvSelector) {
     this.guiceBerryEnvSelector = guiceBerryEnvSelector;
   }
 
-  public Statement apply(final Statement base, final FrameworkMethod method, final Object target) {
+  public Statement apply(Statement base, FrameworkMethod method, Object target) {
+    return apply(base, guiceBerryEnvSelector,
+        new TestDescription(target,target.getClass().getName() + "." + method.getName()));
+  }
+
+  static Statement apply(
```

Comment:
I suspect you're right about vestiges, but I didn't dig through the history.

I've refactored this into a single apply() method which also builds the test descriptions.

One helper method makes sense, since it's common code.  I don't see any need for two.